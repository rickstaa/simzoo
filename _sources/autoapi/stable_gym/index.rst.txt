stable_gym
==========

.. py:module:: stable_gym

.. autoapi-nested-parse::

   Entry_point that register the Stable Gym gymnasium environments.



Subpackages
-----------

.. toctree::
   :maxdepth: 1

   /autoapi/stable_gym/common/index
   /autoapi/stable_gym/envs/index


Submodules
----------

.. toctree::
   :maxdepth: 1

   /autoapi/stable_gym/version/index


Attributes
----------

.. autoapisummary::

   stable_gym.__version__
   stable_gym.__version_tuple__
   stable_gym.ENVS


Classes
-------

.. autoapisummary::

   stable_gym.MaxEpisodeStepsInjectionWrapper


Package Contents
----------------

.. py:class:: MaxEpisodeStepsInjectionWrapper(env)

   Bases: :py:obj:`gymnasium.Wrapper`


   A gymnasium wrapper that injects the ``max_episode_steps`` attribute of the
   ``TimeLimit`` wrapper into the base environment as the
   ``_time_limit_max_episode_steps`` attribute. If the environment is not wrapped in
   a ``TimeLimit`` wrapper, then the ``_time_limit_max_episode_steps`` attribute is
   set to ``None``.

   Wrap a gymnasium environment.
   :param env: The gymnasium environment.
   :type env: gym.Env


   .. py:attribute:: max_episode_steps


.. py:data:: __version__
   :value: '0.20.7'


.. py:data:: __version_tuple__

.. py:data:: ENVS

