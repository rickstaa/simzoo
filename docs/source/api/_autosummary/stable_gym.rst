stable\_gym
===========

.. automodule:: stable_gym
  
   
   
   

   
   
   

   
   
   

   
   
   



.. rubric:: Modules

.. autosummary::
   :toctree:
   :template: custom-module-template.rst
   :recursive:

   stable_gym.common
   stable_gym.envs

