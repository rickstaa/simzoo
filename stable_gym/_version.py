# Stores the package version number so that it can be accessed from other modules.
version = "0.6.4"
version_tuple = version.split(".")
