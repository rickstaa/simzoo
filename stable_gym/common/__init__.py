"""Contains several functions that can be used across all the Stable Gym gymnasium
environments.
"""
from stable_gym.common.disturber import Disturber  # noqa: F401
