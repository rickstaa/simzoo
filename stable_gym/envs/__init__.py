"""The Stable Gym gymnasium environments."""
