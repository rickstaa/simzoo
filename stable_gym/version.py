# Stores the package version number so that it can be accessed from other modules.
__version__ = "0.6.38"
__version_tuple__ = __version__.split(".")
