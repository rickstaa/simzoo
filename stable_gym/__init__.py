"""Module that register the Stable Gym gymnasium environments."""
import gymnasium as gym
from gymnasium.envs.registration import register

# Make module version available.
from .version import __version__, __version_tuple__

# Available environments.
# TODO: Update reward thresholds.
ENVS = {
    "Oscillator-v1": {
        "module": "stable_gym.envs.biological.oscillator.oscillator:Oscillator",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "OscillatorComplicated-v1": {
        "module": "stable_gym.envs.biological.oscillator_complicated.oscillator_complicated:OscillatorComplicated",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "CartPoleCost-v1": {
        "module": "stable_gym.envs.classic_control.cartpole_cost.cartpole_cost:CartPoleCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "Ex3EKF-v1": {
        "module": "stable_gym.envs.classic_control.ex3_ekf.ex3_ekf:Ex3EKF",
        "max_step": 400,
        "reward_threshold": 300,
    },
    "AntCost-v1": {
        "module": "stable_gym.envs.mujoco.ant_cost.ant_cost:AntCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "HalfCheetahCost-v1": {
        "module": "stable_gym.envs.mujoco.half_cheetah_cost.half_cheetah_cost:HalfCheetahCost",
        "max_step": 200,
        "reward_threshold": 300,
    },
    "HopperCost-v1": {
        "module": "stable_gym.envs.mujoco.hopper_cost.hopper_cost:HopperCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "SwimmerCost-v1": {
        "module": "stable_gym.envs.mujoco.swimmer_cost.swimmer_cost:SwimmerCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
    "Walker2dCost-v1": {
        "module": "stable_gym.envs.mujoco.walker2d_cost.walker2d_cost:Walker2dCost",
        "max_step": 250,
        "reward_threshold": 300,
    },
}

for env, val in ENVS.items():
    if env not in gym.envs.registry:  # NOTE: Required because we use namespace packages
        register(
            id=env,
            entry_point=val["module"],
            max_episode_steps=val["max_step"],
            reward_threshold=val["reward_threshold"],
        )
