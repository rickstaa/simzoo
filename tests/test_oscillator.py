"""Test if the Oscillator environment still behaves like the previous Oscillator
environment.
"""

import gymnasium as gym
from gymnasium.logger import ERROR

import stable_gym  # noqa: F401

gym.logger.set_level(ERROR)


class TestOscillator:
    env = gym.make("Oscillator")

    def test_reset(self, snapshot):
        """Test if reset is still equal to the last snapshot."""
        observation, info = self.env.reset(seed=42)
        assert (observation == snapshot).all()
        assert info == snapshot

    def test_step(self, snapshot):
        """Test if steps behave the same."""
        # Perform steps and check if observations are equal.
        self.env.action_space.seed(42)
        for _ in range(5):
            action = self.env.action_space.sample()
            observation, reward, terminated, truncated, info = self.env.step(action)
            assert (observation == snapshot).all()
            assert reward == snapshot
            assert terminated == snapshot
            assert truncated == snapshot
            assert info == snapshot
