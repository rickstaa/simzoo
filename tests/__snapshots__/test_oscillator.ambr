# serializer version: 1
# name: TestOscillator.test_reset
  3.8697802427798167
# ---
# name: TestOscillator.test_reset.1
  2.1943921987602617
# ---
# name: TestOscillator.test_reset.2
  4.292989599556912
# ---
# name: TestOscillator.test_reset.3
  3.4868401452968194
# ---
# name: TestOscillator.test_reset.4
  0.47088673943824766
# ---
# name: TestOscillator.test_reset.5
  4.878111758183779
# ---
# name: TestOscillator.test_reset.6
  8.0
# ---
# name: TestOscillator.test_reset.7
  -4.513159854703181
# ---
# name: TestOscillator.test_reset.8
  dict({
    'reference': 8.0,
    'reference_constraint_error': -16.51315985470318,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -4.513159854703181,
    'state_of_interest': 3.4868401452968194,
  })
# ---
# name: TestOscillator.test_step
  6.054702492845994
# ---
# name: TestOscillator.test_step.1
  1.3536723708128777
# ---
# name: TestOscillator.test_step.10
  False
# ---
# name: TestOscillator.test_step.11
  dict({
    'reference': 8.219875313546899,
    'reference_constraint_error': -16.10320542457622,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -4.323080738123117,
    'state_of_interest': 3.896794575423781,
  })
# ---
# name: TestOscillator.test_step.12
  7.115191374657359
# ---
# name: TestOscillator.test_step.13
  0.0
# ---
# name: TestOscillator.test_step.14
  12.87923968774058
# ---
# name: TestOscillator.test_step.15
  4.631739299753713
# ---
# name: TestOscillator.test_step.16
  0.9626996889912394
# ---
# name: TestOscillator.test_step.17
  6.316237850683443
# ---
# name: TestOscillator.test_step.18
  8.439533636705194
# ---
# name: TestOscillator.test_step.19
  -3.8077943369514813
# ---
# name: TestOscillator.test_step.2
  8.5017041586181
# ---
# name: TestOscillator.test_step.20
  14.499297712519772
# ---
# name: TestOscillator.test_step.21
  False
# ---
# name: TestOscillator.test_step.22
  False
# ---
# name: TestOscillator.test_step.23
  dict({
    'reference': 8.439533636705194,
    'reference_constraint_error': -15.368260700246287,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -3.8077943369514813,
    'state_of_interest': 4.631739299753713,
  })
# ---
# name: TestOscillator.test_step.24
  8.62728249535218
# ---
# name: TestOscillator.test_step.25
  2.931902989654285
# ---
# name: TestOscillator.test_step.26
  7.930094036684647
# ---
# name: TestOscillator.test_step.27
  5.492265561713667
# ---
# name: TestOscillator.test_step.28
  0.904937707651765
# ---
# name: TestOscillator.test_step.29
  7.997941929680929
# ---
# name: TestOscillator.test_step.3
  3.896794575423781
# ---
# name: TestOscillator.test_step.30
  8.6587581932296
# ---
# name: TestOscillator.test_step.31
  -3.1664926315159327
# ---
# name: TestOscillator.test_step.32
  10.026675585444696
# ---
# name: TestOscillator.test_step.33
  False
# ---
# name: TestOscillator.test_step.34
  False
# ---
# name: TestOscillator.test_step.35
  dict({
    'reference': 8.6587581932296,
    'reference_constraint_error': -14.507734438286333,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -3.1664926315159327,
    'state_of_interest': 5.492265561713667,
  })
# ---
# name: TestOscillator.test_step.36
  6.77540453458308
# ---
# name: TestOscillator.test_step.37
  1.2221184287656546
# ---
# name: TestOscillator.test_step.38
  11.808575489982447
# ---
# name: TestOscillator.test_step.39
  6.543094827267196
# ---
# name: TestOscillator.test_step.4
  0.7937362868735947
# ---
# name: TestOscillator.test_step.40
  1.3197459235373448
# ---
# name: TestOscillator.test_step.41
  8.786880459769616
# ---
# name: TestOscillator.test_step.42
  8.877332634950129
# ---
# name: TestOscillator.test_step.43
  -2.334237807682933
# ---
# name: TestOscillator.test_step.44
  5.448666142816425
# ---
# name: TestOscillator.test_step.45
  False
# ---
# name: TestOscillator.test_step.46
  False
# ---
# name: TestOscillator.test_step.47
  dict({
    'reference': 8.877332634950129,
    'reference_constraint_error': -13.456905172732803,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -2.334237807682933,
    'state_of_interest': 6.543094827267196,
  })
# ---
# name: TestOscillator.test_step.48
  7.1504489432514395
# ---
# name: TestOscillator.test_step.49
  4.2907152128944634
# ---
# name: TestOscillator.test_step.5
  5.272303388621858
# ---
# name: TestOscillator.test_step.50
  9.936918693702047
# ---
# name: TestOscillator.test_step.51
  7.234573863164457
# ---
# name: TestOscillator.test_step.52
  1.4361001167276088
# ---
# name: TestOscillator.test_step.53
  10.149039710580631
# ---
# name: TestOscillator.test_step.54
  9.095041255281616
# ---
# name: TestOscillator.test_step.55
  -1.860467392117159
# ---
# name: TestOscillator.test_step.56
  3.461338917131223
# ---
# name: TestOscillator.test_step.57
  False
# ---
# name: TestOscillator.test_step.58
  False
# ---
# name: TestOscillator.test_step.59
  dict({
    'reference': 9.095041255281616,
    'reference_constraint_error': -12.765426136835544,
    'reference_constraint_position': 20.0,
    'reference_constraint_violated': False,
    'reference_error': -1.860467392117159,
    'state_of_interest': 7.234573863164457,
  })
# ---
# name: TestOscillator.test_step.6
  8.219875313546899
# ---
# name: TestOscillator.test_step.7
  -4.323080738123117
# ---
# name: TestOscillator.test_step.8
  18.689027068331118
# ---
# name: TestOscillator.test_step.9
  False
# ---
